addon/components/frost-toggle-button.js

```diff
@@ -0,0 +1,106 @@
+import Ember from 'ember'
+const {computed, Component, ViewUtils} = Ember
+import layout from '../templates/components/frost-toggle-button'
+import FrostEventsProxy from '../mixins/frost-events-proxy'
+
+export default Component.extend(FrostEventsProxy, {
+
+  // ==========================================================================
+  // Properties
+  // ==========================================================================
+
+  layout: layout,
+
+  classNames: ['frost-toggle-button'],
+
+  attributeBindings: [
+    'toggled',
+    'disabled'
+  ],
+
+  classNameBindings: [
+    'disabled'
+  ],
+
+  size: 'medium',
+  toggled: false,
+
+  // ==========================================================================
+  // Functions
+  // ==========================================================================
+
+  init () {
+    this._super(...arguments)
+
+    const toggled = this.toggled
+    Ember.assert(`The value passed to ${this.toString()}{toggled} must be boolean`,
+      typeof toggled === 'boolean' || 'undefined')
+    if (toggled) {
+      this.set('toggled', true)
+    }
+  },
+
+  // ==========================================================================
+  // Computed Properties
+  // ==========================================================================
+
+  // UX requirement - label need hold the value
+  _onLabel: computed('onLabel', function () {
```

Comment:
This is something else that could be handled by PropTypes - getDefaultProps https://github.com/ciena-blueplanet/ember-prop-types#default-property-values